addons/hr_timesheet/static/src/components/task_with_hours/task_with_hours.js

```diff
@@ -1,8 +1,13 @@
 import { registry } from "@web/core/registry";
-import { Many2OneField, many2OneField } from "@web/views/fields/many2one/many2one_field";
-import { onWillStart } from "@odoo/owl";
+import { computeM2OProps, Many2One } from "@web/views/fields/many2one/many2one";
+import { buildM2OFieldDescription, Many2OneField } from "@web/views/fields/many2one/many2one_field";
+import { Component, onWillStart } from "@odoo/owl";
+
+export class TaskWithHours extends Component {
```

Comment:
Maybe someone already asks that, if yes sorry. 🙈 

Does it not better to extends `Many2oneField` component? By doing that, you can call super in `m2oProps` instead of doing the same think the get the "standardProps" for the new component. 🙂 

(this comment is the same for the other many2one components) 😇

Reply:
Also by doing that, we can also use the template of the Many2oneField component. 🙂

Reply:
Hello, I wanted the field components to be independent from each other.
At first, the widgets were more independent than it is now, each field defined its own attributes / options / props because the widgets don't need every option `Many2OneField` has. So there was no `computeM2OProps`, no `Many2OneField.props`, no `extractM2OFieldProps`, no `buildM2OFieldDescription`. The fields did their own job.
I don't prevent extending component but I think it's better to compose multiple components than extending an existing one.